docs/redesign-proposal.md

```diff
@@ -0,0 +1,93 @@
+# Navigator design
+
+Navigator as a system intends to provide a way to easily provision, monitor and
+manage complex systems on top of cluster schedulers, initially targetting
+Kubernetes and OpenShift.
+
+## Architecture
+
+All communication between components of the Navigator system should communicate
+through the `navigator-apiserver`. This allows for a consistent audit trail,
+recorded in a central place. Monitoring services such as `kube-state-metrics`
+can then be used to track changes to the contents of the API server to be
+consumed by something like Prometheus.
+
+## Pilots
+
+Databases run by Navigator are wrapped by helper processes called 'Pilots'.
+
+Pilots contain the operational knowledge needed to operate their respective
+databases with the assistance of the `navigator-apiserver`.
+
+Each Pilot runs as PID 1 within the main application container of the
+application it 'manages'. This allows us to closely and carefuly monitor the
+lifecycle events and signals that the subprocess receives. This will allow us
+to run Pilot in environments other than Kubernetes, despite the individual
+properties of how these systems work.
+
+Because the Pilot runs before the actual application has started, it is able to
+perform any preflight or configuration checks at an appropriate time. This may
+include waiting for other Pilots to start, or determining a leader (with the
+help of the `navigator-apiserver`) that should begin the bootstrapping of a new
+cluster for that application. This also applies at exit time. The Pilot is able
+to trap signals sent to the container and ensure the underlying application
+exits cleanly, and if it doesn't, report this diagnostic information back to
+the `navigator-apiserver`.
+
+Pilots should watch the `navigator-apiserver` for changes to resources it
+requires, and use the apiserver to communicate with other instances of Pilot
+within the same application cluster (eg. within the same Elasticsearch
+cluster). This may include things like notifications that a scale down event
+has been requested, or even used to elect a 'leader' between the Pilots in the
+event one is required for the proper functioning of that application.
+
+Each Pilot should also be able to expose a metrics endpoint to be scraped by
+systems like Prometheus. This could be implemented via some of the existing
+exporters for common applications, or directly within the Pilot itself if
+required.
+
+## navigator-controller-manager
```

Comment:
I think we should we mention something around helm/tiller integration points here